src/main/java/mk/edu/uklo/fikt/fiktexamweb/controller/SubjectController.java

```diff
@@ -1,58 +1,88 @@
 package mk.edu.uklo.fikt.fiktexamweb.controller;
 
+import java.security.Principal;
 import java.util.List;
 
+import javax.jws.soap.SOAPBinding;
 import javax.validation.Valid;
 
 import mk.edu.uklo.fikt.fiktexamweb.DTO.SubjectTopics;
+import mk.edu.uklo.fikt.fiktexamweb.DTO.TeacherSubjects;
 import mk.edu.uklo.fikt.fiktexamweb.model.Topic;
-import mk.edu.uklo.fikt.fiktexamweb.util.TopicBL;
+import mk.edu.uklo.fikt.fiktexamweb.model.User;
+import mk.edu.uklo.fikt.fiktexamweb.util.TopicService;
+import mk.edu.uklo.fikt.fiktexamweb.util.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
 
 import mk.edu.uklo.fikt.fiktexamweb.model.Subject;
-import mk.edu.uklo.fikt.fiktexamweb.model.User;
-import mk.edu.uklo.fikt.fiktexamweb.util.SubjectBL;
+import mk.edu.uklo.fikt.fiktexamweb.util.SubjectService;
 
-@RestController
+@Controller
 @RequestMapping({"/subject"})
+@SessionAttributes
 public class SubjectController {
 	
 	@Autowired
-	SubjectBL subjectBl;
+	private SubjectService subjectService;
 
 	@Autowired
-	TopicBL topicBL;
-	
-	@PreAuthorize("hasAnyRole('Admin')")
+	private TopicService topicService;
+
+	@Autowired
+	private UserService userService;
+
+	//go to P2 screen which contains all subjects for authencicated teacher
+	@GetMapping("/subjectscreen")
+	public String getSubjects(Model model){
+		getByTeacher(userService.getIdByUsername(
+				SecurityContextHolder.getContext().getAuthentication().getName()),
+				model);
+		Subject subject = new Subject();
+		model.addAttribute("subject", subject);
+		return "P2";
+	}
+
+	//go to S2 screen which contains all subjects for student to select on which subject is the test
+	@GetMapping("/studentsubjectscreen")

```

Comment:
use hyphens to separate the path by words, something like: `student-subject-screen`